### ISSUE TITLE
total-functions/no-array-destructuring false positive with string literals

### ISSUE THREAD

**Issue by awwit** [danielnixon/eslint-plugin-total-functions#37]
The code that reproduces the error:

```ts
interface Response {
  'error-code': number
}

function getError(
  { 'error-code': errorCode }: Response // <-- total-functions/no-array-destructuring
): string {
  return errorCode.toString()
}
```

There should be no `total-functions/no-array-destructuring` in this code.
For normal properties (not quoted), everything works fine.

I am using the library version:
- `eslint@7.5.0`
- `eslint-plugin-total-functions@1.38.0`

Eslint config:

```js
extends: [
  ...,
  'plugin:total-functions/recommended',
],
plugins: [
  ...,
  'total-functions'
]
```

Without rewriting the rules for `eslint-plugin-total-functions`.

**Comment by danielnixon** [comment]
Thanks for reporting @awwit . I'll fix this soon. In the meantime, PRs encouraged :P

**Comment by danielnixon** [comment]
Released as v1.39.0. Please let me know how you go @awwit

**Comment by awwit** [comment]
@danielnixon I confirm that the error has been fixed.
Thanks! =)

**Comment by danielnixon** [comment]
Yikes. Hold tight.

**Comment by danielnixon** [comment]
@awwit let me know how v1.40.0 goes. Sorry for the churn.
